pkg/kubelet/cri/streaming/portforward/httpstream.go

```diff
@@ -252,11 +261,24 @@ func (h *httpStreamHandler) portForward(p *httpStreamPair) {
 	err := h.forwarder.PortForward(ctx, h.pod, h.uid, int32(port), p.dataStream)
 	klog.V(5).InfoS("Connection request done invoking forwarder.PortForward for port", "connection", h.conn, "request", p.requestID, "port", portString)
 
-	if err != nil {
-		msg := fmt.Errorf("error forwarding port %d to pod %s, uid %v: %v", port, h.pod, h.uid, err)
-		utilruntime.HandleError(msg)
-		fmt.Fprint(p.errorStream, msg.Error())
+	// happy path, we have successfully completed forwarding task
+	if err == nil {
+		return
 	}
+
+	// forwarding in one direction has been completed.
+	// other direction: we got RST, so we should reset data && err stream
+	if errors.Is(err, syscall.EPIPE) || errors.Is(err, syscall.ECONNRESET) {
+		klog.V(5).InfoS("done invoking forwarder.PortForward but got EPIPE", "conn", h.conn, "request", p.requestID, "port", portString)
```

Comment:
Can this klog message say which one it got: EPIPE or ECONNRESET?

Reply:
Thanks for the reminder, it has been added.